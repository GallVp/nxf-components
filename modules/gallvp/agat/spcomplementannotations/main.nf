process AGAT_SPCOMPLEMENTANNOTATIONS {
    tag "${meta.id}"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'https://depot.galaxyproject.org/singularity/agat:1.4.2--pl5321hdfd78af_0'
        : 'biocontainers/agat:1.4.2--pl5321hdfd78af_0'}"

    input:
    tuple val(meta), path(ref_gff)
    tuple val(meta2), path(add_gffs)
    path config

    output:
    tuple val(meta), path("*.gff"), emit: gff
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def config_param = config ? "--config ${config}" : ''
    def file_names = "${add_gffs}".split(' ')
    def add_gff_param = file_names.collect { "--add ${it}" }.join(' ')
    if (file_names.contains("${prefix}.gff") || "${ref_gff}" == "${prefix}.gff") {
        error("Input and output names are the same, use \"task.ext.prefix\" to disambiguate!")
    }
    """
    agat_sp_complement_annotations.pl \\
        --ref ${ref_gff} \\
        ${add_gff_param} \\
        ${config_param} \\
        ${args} \\
        --output ${prefix}.gff

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        agat: \$(agat_sp_complement_annotations.pl -h | sed -n 's/.*(AGAT) - Version: \\(.*\\) .*/\\1/p')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def file_names = "${add_gffs}".split(' ')
    if (file_names.contains("${prefix}.gff") || "${ref_gff}" == "${prefix}.gff") {
        error("Input and output names are the same, use \"task.ext.prefix\" to disambiguate!")
    }
    """
    touch ${prefix}.gff

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        agat: \$(agat_sp_complement_annotations.pl -h | sed -n 's/.*(AGAT) - Version: \\(.*\\) .*/\\1/p')
    END_VERSIONS
    """
}
