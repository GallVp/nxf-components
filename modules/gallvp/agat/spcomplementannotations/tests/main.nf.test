nextflow_process {

    name "Test Process AGAT_SPCOMPLEMENTANNOTATIONS"
    script "../main.nf"
    process "AGAT_SPCOMPLEMENTANNOTATIONS"

    tag "modules"
    tag "modules_gallvp"
    tag "agat"
    tag "agat/spcomplementannotations"

    test("candidatus_portiera_aleyrodidarum - Gffs") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/candidatus_portiera_aleyrodidarum/genome/gff/test1.gff', checkIfExists: true)
                ]
                input[1] = [
                    [],
                    [
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/candidatus_portiera_aleyrodidarum/genome/gff/test2.gff', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/candidatus_portiera_aleyrodidarum/genome/gff/test3.gff', checkIfExists: true)
                    ]
                ]
                input[2] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    file(process.out.gff[0][1]).readLines()[0..2]
                    ).match()
                }
            )
        }

    }

    test("candidatus_portiera_aleyrodidarum - Gffs - stub") {

        options '-stub'

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/candidatus_portiera_aleyrodidarum/genome/gff/test1.gff', checkIfExists: true)
                ]
                input[1] = [
                    [],
                    [
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/candidatus_portiera_aleyrodidarum/genome/gff/test2.gff', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/prokaryotes/candidatus_portiera_aleyrodidarum/genome/gff/test3.gff', checkIfExists: true)
                    ]
                ]
                input[2] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}