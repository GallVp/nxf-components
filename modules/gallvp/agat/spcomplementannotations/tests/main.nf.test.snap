{
    "candidatus_portiera_aleyrodidarum - Gffs": {
        "content": [
            [
                "versions.yml:md5,3b461a98eb18641f842397b7c626f59c"
            ],
            [
                "##gff-version 3",
                "##sequence-region NC_018507.1 1 358242",
                "NC_018507.1\tAGAT\tgene\t1\t1926\t.\t+\t.\tID=agat-gene-1;Name=dnaK;db_xref=COG:COG0443;gene=dnaK;inference=ab initio prediction:Prodigal:002006,similar to AA sequence:UniProtKB:O87712;locus_tag=JKHLNHAL_00001;product=Chaperone protein DnaK"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-06T10:38:15.349104"
    },
    "candidatus_portiera_aleyrodidarum - Gffs - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.gff:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3b461a98eb18641f842397b7c626f59c"
                ],
                "gff": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.gff:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3b461a98eb18641f842397b7c626f59c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-06T10:13:11.651559"
    }
}